# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../kiss/posix-fixes.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---


This is mainly needed when used with dietlibc - but also more portable
and more secure.

  Rene Rebe <rene@exactcode.de>

diff -ur kiss-0.21/src/doread.c src.kiss.-1.25906.3396530713/kiss-0.21/src/doread.c
--- kiss-0.21/src/doread.c	1998-09-25 14:07:09.000000000 +0200
+++ kiss-0.21/src/doread.c	2003-08-17 23:54:10.000000000 +0200
@@ -17,11 +17,11 @@
 	for (cp = s.str [1]; *cp; cp++)
 	    if (! isupper (*cp))
 		return (warning ("variable must be in upper case"));
-	gets (buf);
+	fgets (buf, FILENAMELEN, stdin);
 	addtoenv (s.str [1], buf);
     }
     else
-	gets (buf);
+	fgets (buf, FILENAMELEN, stdin);
     
     return (0);
 }
diff -ur kiss-0.21/src/runcmd.c src.kiss.-1.25906.3396530713/kiss-0.21/src/runcmd.c
--- kiss-0.21/src/runcmd.c	1998-09-25 14:07:10.000000000 +0200
+++ kiss-0.21/src/runcmd.c	2003-08-17 23:11:35.000000000 +0200
@@ -1,5 +1,7 @@
 #include "kiss.h"
 
+#include <getopt.h>
+
 /* define DEBUG_ONE_CMD to get one shot at gdb'ing the child process */
 /* then make a -g version of the command to see and set a trace in the code */
 
